### PR TITLE
feat(query): add support for changes function for subqueries

### PR DESCRIPTION
**Pull request by kvpetrov**
**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

Breaking changes may include:
- Any schema changes to any Cassandra tables
- The serialized format for Dataset and Column (see .toString methods)
- Over the wire formats for Akka messages / case classes
- Changes to the HTTP public API
- Changes to query parsing / PromQL parsing

**Other information**:

### CODE REVIEW

**Review by vishramachandran — APPROVED** [review]

File: query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala [hunk 1]
```diff
@@ -143,6 +143,34 @@ class SumOverTimeFunction(var sum: Double = Double.NaN, var count: Int = 0) exte
...
+    sampleToEmit: TransientRow,
+    queryConfig: QueryConfig
+  ): Unit = {
+    var lastValue = Double.NaN
+    var changes = 0
+    if (window.size > 0) lastValue = window.head.getDouble(1)
+    var i = 1;
+    while (i < window.size) {
+      val curValue = window.apply(i).getDouble(1)
+      if (!(curValue.isNaN && curValue.isNaN)) {
```
[the remark this answers is not in the document]
Comment (kvpetrov) [reply query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala]: good catch, indeed it was a typo (it did not pop up in the manual testing), added the unit tests covering the edge cases.
